io/io/src/TStreamerInfo.cxx

```diff
@@ -5678,6 +5678,12 @@ TVirtualStreamerInfo *TStreamerInfo::GenerateInfoForPair(const std::string &firs
    // This TStreamerInfo is then used as if it was read from a file to generate
    // and emulated TClass.
 
+   if (hint_pair_offset && hint_pair_offset == hint_pair_size) {
```

Comment:
I really meant to please add an assert on `(!hint_pair_offset == !hint_pair_size)`: it's checking that either both or non are `==0`. Your current `if` doesn't check that.